src/libws_private.c

```diff
@@ -792,9 +792,8 @@ void ws_read_callback(struct bufferevent *bev, void *ptr)
 		switch ((state = _ws_read_server_handshake_reply(ws, in)))
 		{
 			case WS_PARSE_STATE_ERROR:
-				// TODO: Do anything else here?
-				_ws_shutdown(ws);
-				break;
+                ws_close_with_status(ws, WS_CLOSE_STATUS_PROTOCOL_ERR_1002);

```

Comment:
Not 100% this is the way to do it, since we haven't completed the server handshake? But maybe you have tried this for real. Anyway I don't see any issue with doing it like this unless it crashes something of course :)